package.json

```diff
@@ -236,9 +235,10 @@
     "@types/node": "^8.10.20",
     "@types/prop-types": "^15.5.3",
     "@types/puppeteer": "^1.6.2",
-    "@types/react": "^16.3.14",
+    "@types/react": "16.3.14",
```

Comment:
Problem here: by using `^` for this package, installing another `@types/*` package that depends on `@types/react` using the `*` semver range makes us upgrade to the latest version that our ranges allow, so installing `@types/react-router-dom` auto-upgraded this package to 16.4.x, and that version breaks a lot of our types stuff since we still use React 16.3. 

Switching to `~` didn't help, because even upgrading to 16.3.18 breaks tons of our kibana typescript stuff. I don't think the `@types/*` packages really follow semver the same way as other packages do, similarly to how linting rulesets can't really follow semver, because every single change is most likely breaking in some way.

tl;dr since newer versions of `@types/react` cause lots of typescript errors in kibana, we should be explicit about the version that works with kibana and specify `16.3.14` directly.

Reply:
Makes sense! Let's just double check with @spalger or @epixa to see if they have any objections.

Reply:
Sounds good, I'm going to merge this now since it's not actually changing anything with the real deps, and if there are objections or considerations we can revert the package.json change later if we need to. Thanks!

Reply:
No objections. It seems strange that it breaks so much though.